### DISCUSSION TITLE
I want to support Solana chain in Rainbow. I have added Phantom wallet in Rainbow. Is there any reference case?

### DISCUSSION THREAD

**Discussion by maicFir** [rainbow-me/rainbowkit#2024] [Q&A]
![Snipaste_2024-06-03_20-38-11](https://github.com/rainbow-me/rainbowkit/assets/21973334/1fa0f4e7-541b-4fe5-b0a7-e9a09dfc5626)
Unable to switch solna chain
![image](https://github.com/rainbow-me/rainbowkit/assets/21973334/9a6c9944-669a-47b5-892f-9f05a6e169b2)

**Comment by 0xmkh** [comment]
@maicFir We don't support solana chain with Phantom wallet. Is this something you'd like us to look at ?

**Reply by maicFir** [reply]
Yes, I want to integrate the solna chain in Rainbow. I want to know if there are other examples or better solutions and ideas.thank you very much

**Reply by 0xmkh** [reply]
Yeah i'll let you know what i come up with

**Comment by misterjame** [comment]
+1 on this for solana support, it seems the only people doing this are dynamic xyz and they are closed source. Currently we are using rainbow kit with the @solana libraries it doesn't have the polish of just regular rainbow kit. Solana is getting pretty popular I think it would be good to add support (both in rainbow wallet and rainbowkit), Metamask has already done so.

**Reply by DanielSinclair** [reply]
We hear you, but we will need to wait until interoperability improves before this will be feasible. Solana and EVM are still fully divergent tech stacks, and we won't see direct Solana support from Wagmi and the underlying RainbowKit dependencies anytime soon. Your best bet today to cross this chasm will be to build your dApp around chain abstraction protocols like Socket, Relay, or LayerZero that allow EVM wallets to interact with Solana indirectly.

**Reply by misterjame** [reply]
Sounds nice, but it's at great risk to your project (which I am quite a fan of). Phantom is already supporting EVM, Solana, and BTC. If Phantom makes PhantomKit, then the whole project is at risk of being overshadowed. 

Anyway, thank you for the answer.
